Sources/AppStoreConnectClient/AppStoreConnectClient.swift

```diff
@@ -77,14 +79,95 @@ public struct AppStoreConnectClient {
         default:
             let error = try handleErrors(from: response)
             throw error
         }
         return result
     }
     
+    /// Fetches a collection of TestFlight builds for a specified app from the App Store Connect API.
+    /// end-point: v1/builds
+    /// - Parameter app: The app for which to fetch builds.
+    /// - Returns: An array of `Build` objects.
+    /// - Throws: An error of type `AppStoreConnectError` if the fetch operation fails.
+    func fetchBuilds(for app: Application) async throws -> [Build] {
+        let response = try await client.builds_hyphen_get_collection(
+            query: .init(
+                filter_lbrack_app_rbrack_: [app.id],
+                sort: Operations.builds_hyphen_get_collection.Input.Query.sortPayload?.init([._hyphen_version]),
+                fields_lbrack_builds_rbrack_: Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_Payload?.init(
+                    [.version, .uploadedDate, .minOsVersion]
+                )
+            )
+        )
+        let result: [Build]
+        var responseJson: Components.Schemas.BuildsResponse
+        switch response {
+        case .ok(let okResponse):
+            responseJson = try okResponse.body.json
+        default:
+            let error = try handleErrors(from: response)
+            throw error
+        }
+        result = responseJson.data.compactMap { Build(schema: $0) }
+        return result
+    }
+    
+    /// Fetches the pre-release version associated with a specific TestFlight's build from the App Store Connect API.
+    /// end-point: v1/builds/{id}/preReleaseVersion
+    /// - Parameter id: The build-id for which to fetch the pre-release version.
+    /// - Returns: A `PreReleaseVersion` object.
+    /// - Throws: `PreReleaseVersionError.idIsNil` if the build's id is nil, 
+    /// or an error of type `AppStoreConnectError` if the fetch operation fails.
+    func fetchPreReleaseVersion(by id: Build) async throws -> PreReleaseVersion {
+        guard let valueId = id.id else {
+            throw PreReleaseVersionError.idIsNil
+        }
+        let response = try await client.builds_hyphen_preReleaseVersion_hyphen_get_to_one_related(
+            path: .init(id: valueId)
+        )
+        var responseJson: Components.Schemas.PrereleaseVersionWithoutIncludesResponse
+        switch response {
+        case .ok(let okResponse):
+            responseJson = try okResponse.body.json
+        default:
+            let error = try handleErrors(from: response)
+            throw error
+        }
+        let result = PreReleaseVersion(schema: responseJson)
+        return result
+    }
+    
+    /// Handles error responses returned by the App Store Connect API when fetching app collections.
+    /// - Parameter response: The response received from the API.
+    /// - Throws: An error of type `AppStoreConnectError` if the response indicates an error.
+    private func handleErrors(
```

Comment:
Need try reuse one function for handle error for all responses
For it need look on `Output` if it protocol then you func definition looks like 
```swift
private func handleErrors(
        from response: Output
    ) throws -> AppStoreConnectError
```

Reply:
Changed code of error functions.